src/escalation/mod.rs

```diff
@@ -0,0 +1,32 @@
+#![allow(unused)]
+#[cfg(target_os = "macos")]
+mod darwin;
+mod unix;
+
+use std::process::Command;
+use tokio::process::Command as AsyncCommand;
+
+#[derive(Debug, thiserror::Error)]
+#[allow(dead_code)]
+pub enum Error {
+    #[error("Could not become root! Searched for sudo, doas, su")]
+    UnixNotDetected,
+    #[error("User failed to confirm")]
+    MacOSDenial,
+}
+
+#[cfg(target_os = "linux")]
```

Comment:
this should also be used on FreeBSD and OpenBSD and such

Reply:
that's true, but we'll cross that bridge when we get there, because I have not compiled this for either OS yet.